install/operator/build.sh

```diff
@@ -1,15 +1,40 @@
 #!/bin/bash
-set -e 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
-cd $DIR
-
-echo ===========================================
-echo Running go build via a docker build
-echo ===========================================
-docker build -t syndesis-operator-builder . -f Dockerfile-builder
-
-echo ===========================================
-echo Extracting go binary from the builder image
-echo ===========================================
-docker run syndesis-operator-builder cat /syndesis-operator > syndesis-operator
-chmod a+x syndesis-operator
+#
+# Builds the operator, using operator-sdk for developers that have it
+# installed locally, or using docker if they don't have it installed.
+#
+set -e
+cd $(dirname "${BASH_SOURCE[0]}")
+source "./.lib.sh"
+
+OPERATOR_GO_PACKAGE="github.com/syndesisio/syndesis/install/operator"
+OPERATOR_IMAGE_NAME="$(readopt --image-name         syndesis/syndesis-operator)"
+S2I_STREAM_NAME="$(readopt     --s2i-stream-name    syndesis-operator)"
+OPERATOR_BUILD_MODE="$(readopt --operator-build     auto)"
+IMAGE_BUILD_MODE="$(readopt    --image-build        auto)"
+
+if [[ -n "$(readopt --help)" ]] ; then 
+	cat <<ENDHELP
+
+usage: ./build.sh [options]
+
+where options are:
+  --help                             display this help messages
+  --operator-build <auto|docker|go>  how to build the operator executable (default: auto)
+  --image-build <auto|docker|s2i>    how to build the image (default: auto)
+  --image-name <name>                docker image name (default: syndesis/syndesis-operator)
+  --s2i-stream-name <name>           s2i image stream name (default: syndesis-operator)
+
+ENDHELP
+	exit 0
+fi
+
+#
+# TODO Could we avoid copying these files by moving them under the build directory
```

Comment:
Big kudos for improving this, I also like think that moving `addons` is the way to go. At least for now we could delete the folder after the operator is built